src/schema/sources.ts

```diff
@@ -1245,22 +1254,33 @@ export const resolvers: IResolvers<any, Context> = {
     },
     mySourceMemberships: async (
       _,
-      args: ConnectionArguments,
+      args: SourcesByType,
       ctx,
       info,
     ): Promise<Connection<GQLSourceMember>> => {
+      const { type, ...connectionArgs } = args;
+
       return paginateSourceMembers(
         (queryBuilder, alias) => {
-          return queryBuilder
+          queryBuilder
             .andWhere(`${alias}."userId" = :user`, { user: ctx.userId })
             .andWhere(
               `${
                 graphorm.mappings.SourceMember.fields.roleRank.select as string
               } >= 0`,
             )
             .addOrderBy(`${alias}."createdAt"`, 'DESC');
+
+          if (type) {
+            queryBuilder = queryBuilder
```

Comment:
You don't need to assign, `queryBuilder` methods mutate in place?

Reply:
We do it everywhere else in the file, so I followed the same pattern.

Reply:
Somewhere we do somewhere we don't, we should pick one or the other. Non blocking though for this.